plugins/jobs/src/js/JobStopRun.js

```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+import { componentFromStream, graphqlObservable } from "data-service";
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/switchMap";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/catch";
+import gql from "graphql-tag";
+
+import JobStopRunModal from "./components/JobStopRunModal";
+
+import defaultSchema from "./data/JobModel";
+
+const stopJobRun = gql`
+  mutation {
+    stopJobRun(id: $jobId, jobRunId: $jobRunId) {
+      jobId
+    }
+  }
+`;
+
+function executeStop({ jobId, jobRunId, onSuccess }) {
+  return graphqlObservable(stopJobRun, defaultSchema, {
+    jobId,
+    jobRunId
+  })
+    .map(_ => ({ done: true }))
+    .do(_ => onSuccess())
+    .startWith({ done: false });
+}
+
+function stopOperation() {
+  const stopSubject$ = new Subject();
+  const stop$ = stopSubject$.switchMap(executeStop).catch(error => {
+    return stop$.startWith({
+      errorMsg: error.response.message,
+      done: true
+    });
+  });
+
+  return {
+    stop$,
+    stopHandler: (jobId, jobRunId, onSuccess) => {
+      stopSubject$.next({ jobId, jobRunId, onSuccess });
+    }
+  };
+}
+
+const JobStopRun = componentFromStream(props$ => {
```

Comment:
Can we please add some basic tests for this? I would at least like to test the `if (jobRuns.length === 1) {` line below, see if a mutation goes through and if we can deal with an API failure gracefully

Reply:
I agree.